app/command.go

```diff
@@ -241,11 +252,12 @@ func (a *App) mentionsToTeamMembers(message, teamId string) model.UserMentionMap
 	}
 
 	atMentionMap := make(model.UserMentionMap)
+	wg.Wait()
 
-	for i := 0; i < len(possibleMentions); i++ {
-		if mention := <-mentionChan; mention != nil {
-			atMentionMap[mention.Name] = mention.Id
-		}
+	lengthMentions := len(mentionChan)
```

Comment:
Instead of iterating over the length, you can just range over the channel mentions directly:
```go
for mention := range mentionChan {
    atMentionMap[mention.Name] = mention.Id
}
```

Reply:
Much cleaner, thanks for the suggestion!

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/13762/commits/9c01498c3cde0eac0ad1882bbf9f66459083f19b.